libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java

```diff
@@ -255,7 +255,8 @@ public void run() {
                     updateSync(sync, SyncState.Status.DONE, 100, callback);
                 } catch (RestClient.RefreshTokenRevokedException re) {
                     SmartSyncLogger.e(TAG, "Exception thrown in runSync", re);
-                    // Do not do anything - let the logout go through!
+                    // Failed due to invalid refresh token. Update status to failed.
+                    updateSync(sync, SyncState.Status.FAILED, UNCHANGED, callback);
```

Comment:
We send out an intent when the refresh token is revoked. The app could have a listener for this intent and cancel any in-flight sync. @narakim 

Here's the action on the intent:
```
public static final String ACCESS_TOKEN_REVOKE_INTENT = "access_token_revoked";
```

Look in `ClientManager` for more details.